### PR TITLE
Harvester of Souls implementation

### PR DESCRIPTION
**Pull request by Neicigam**
Since HoS can create multiple copies of a single card, cards must all have a unique identifier now. I've implemented this and made a few changes to keep current features working. I have probably missed many other details, I'll dive again into all the files tomorrow.

### CODE REVIEW

File: src/components/DeckBuilderDryRunView/index.js [hunk 1]
```diff
@@ -224,7 +224,7 @@ class DeckBuilderDryRunView extends React.Component {
   containsFreeze = deck => {
     const deckIds = deck.map(card => card.id)
     const freezeCards = ['W1', 'W2', 'W4', 'W6', 'W8', 'W11']
-    return deckIds.some(id => freezeCards.includes(id))
+    return deckIds.some(id => freezeCards.includes(id.split('#')[0]))
```
Comment (KittySparkles) [inline src/components/DeckBuilderDryRunView/index.js:227]: I can see how this is going to be absolute hell quite quickly; I’m not sure I want us to have to split the ID everywhere we want to use it. Maybe we should handle things a little differently. Perhaps we could have 2 IDs per card: `id`, which is the card ID like right now, and `idx` which is a unique identifier so we can differentiate `W1` from `W1` when necessary. This way, all the components relying on `id` (like `Deck`) still work out of the box, we can differentiate cards within the dry-runner with their `idx`. What do you think?

File: src/components/DeckMechanisms/index.js [hunk 2]
```diff
@@ -358,8 +359,12 @@ export default class DeckMechanisms extends React.Component {
           'T' + arrayRandom([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 13, 14])
         const token = resolveCardForLevel({ id })
         token.level = [5, 6, 6, 8, 10][card.level - 1]
+        token.rarity = null
```
Comment (KittySparkles) [inline src/components/DeckMechanisms/index.js:362]: This has been done on master with https://github.com/KittySparkles/stormbound-kitty/commit/e7fec23415390409b8ab434c5eefe4c51c309a99.

File: src/components/DeckMechanisms/index.js [hunk 3]
```diff
@@ -468,6 +473,35 @@ export default class DeckMechanisms extends React.Component {
         break
       }
 
+      // Harvester of Souls
+      case 'N38': {
+        const id = arrayRandom(
+          cards
+            .filter(card => card.type === 'unit' && card.id !== 'T12')
```
Comment (KittySparkles) [inline src/components/DeckMechanisms/index.js:480]: This has been done on master with https://github.com/KittySparkles/stormbound-kitty/commit/55e609e61d45d4d69414b2225ac52c627dde9703.

File: src/components/DeckMechanisms/index.js [hunk 4]
```diff
@@ -468,6 +473,35 @@ export default class DeckMechanisms extends React.Component {
...
+        copiedCard.id =
+          id +
+          '#' +
+          this.state.deck.filter(card => card.id.split('#')[0] === id).length
+
+        if (copiedCard.token) {
+          if (Math.random() < PROBABILITIES.NO_MOVEMENT_TOKEN) {
+            // Klaxi and Rain of Frogs spawn tokens with no movement
+            copiedCard.movement = 0
+          }
```
Comment (KittySparkles) [inline src/components/DeckMechanisms/index.js:496]: Is that legitimately a thing? Mirz can spawns tokens with no movement?

File: src/components/DeckMechanisms/index.js [hunk 5]
```diff
@@ -468,6 +473,35 @@ export default class DeckMechanisms extends React.Component {
...
+          copiedCard.level = [5, 6, 6, 8, 10][copiedCard.level - 1]
+          copiedCard.rarity = undefined
```
Comment (KittySparkles) [inline src/components/DeckMechanisms/index.js:498]: This is no longer necessary.

File: src/components/DeckMechanisms/index.js (hunk 4 above)
Comment (Neicigam) [reply src/components/DeckMechanisms/index.js:496]: Not Mirz, but Harversters yes. It was discussed in #bugs-and-issues a few days ago

Comment (KittySparkles) [reply src/components/DeckMechanisms/index.js:496]: That is incredibly bad. Fair enough, good catch.

**Pull request closed without merging**